src/builder/pdb.rs

```diff
@@ -0,0 +1,268 @@
+            .with_labels(role_selector_labels.clone())
+            .with_label(
+                APP_MANAGED_BY_LABEL.to_string(),
+                format_full_controller_name(operator_name, controller_name),
+            )
+            .build();
+        Ok(PdbBuilder {
+            metadata,
+            selector: LabelSelector {
+                match_expressions: None,
+                match_labels: Some(role_selector_labels),
+            },
+            ..PdbBuilder::default()
+        })
+    }
+
+    pub fn metadata(self, metadata: impl Into<ObjectMeta>) -> PdbBuilder<ObjectMeta, (), ()> {
+        PdbBuilder {
+            metadata: metadata.into(),
+            selector: self.selector,
+            max_unavailable: self.max_unavailable,
+            min_available: self.min_available,
+            _constraints: self._constraints,
+        }
+    }
+}
+
+impl PdbBuilder<ObjectMeta, (), ()> {
+    pub fn selector(self, selector: LabelSelector) -> PdbBuilder<ObjectMeta, LabelSelector, ()> {
+        PdbBuilder {
+            metadata: self.metadata,
+            selector,
+            max_unavailable: self.max_unavailable,
+            min_available: self.min_available,
+            _constraints: self._constraints,
+        }
+    }
+}
+
+impl PdbBuilder<ObjectMeta, LabelSelector, ()> {
+    pub fn max_unavailable(
+        self,
+        max_unavailable: u16,
+    ) -> PdbBuilder<ObjectMeta, LabelSelector, bool> {
+        PdbBuilder {
+            metadata: self.metadata,
+            selector: self.selector,
+            max_unavailable: Some(max_unavailable),
+            min_available: self.min_available,
+            _constraints: true, // Some dummy value to set Constraints to something other than ()
+        }
+    }
+
+    #[deprecated(
+        since = "0.51.0",
+        note = "It is strongly recommended to use [`max_unavailable`]. Please read the ADR on Pod disruptions before using this function."
+    )]
+    pub fn min_available(self, min_available: u16) -> PdbBuilder<ObjectMeta, LabelSelector, bool> {
+        PdbBuilder {
+            metadata: self.metadata,
+            selector: self.selector,
+            max_unavailable: self.max_unavailable,
+            min_available: Some(min_available),
+            _constraints: true, // Some dummy value to set Constraints to something other than ()
+        }
+    }
+}
+
+impl PdbBuilder<ObjectMeta, LabelSelector, bool> {
+    pub fn build(self) -> PodDisruptionBudget {
+        PodDisruptionBudget {
+            metadata: self.metadata,
+            spec: Some(PodDisruptionBudgetSpec {
+                max_unavailable: self.max_unavailable.map(i32::from).map(IntOrString::Int),
+                min_available: self.min_available.map(i32::from).map(IntOrString::Int),
+                selector: Some(self.selector),
+                /// As this is beta as of 1.27 we can not use it yet,
+                /// so this builder does not offer this attribute.
```

Comment:
Because this feature is still in beta in k8s version 1.27, the builder currently does not offer this attribute.